dbcon/joblist/distributedenginecomm.cpp

```diff
@@ -308,13 +308,24 @@ void DistributedEngineComm::Setup()
 
             writeToLog(__FILE__, __LINE__, "Could not connect to PMS" + std::to_string(connectionId) + ": " + ex.what(), LOG_TYPE_ERROR);
             cerr << "Could not connect to PMS" << std::to_string(connectionId) << ": " << ex.what() << endl;
+            
```

Comment:
These two exception handlers are identical so it would be better to reduce boilerplate and put the handlers into a function.

Reply:
Maybe, maybe not. To create a new function could de-clutter this code. but it would also be a bit klutzy itself. I'm not sure why the newPmCount == 0 code was added. Must of had a good reason.

Reply:
This can be a crash when for some reason there is no more PM to consider.

Reply:
I personally strongly disagree adding a function to reduce boilerplate code shouldn't be done here. However I can tolerate this.